src/lib.rs

```diff
@@ -1,18 +1,28 @@
-#![warn(trivial_casts, trivial_numeric_casts, unused_import_braces)]
+#![deny(trivial_casts, trivial_numeric_casts, unused_import_braces)]
 #![deny(missing_debug_implementations, missing_copy_implementations)]
-#![warn(clippy::expl_impl_clone_on_copy)]
-#![warn(clippy::float_cmp_const)]
-#![warn(clippy::linkedlist)]
-#![warn(clippy::map_flatten)]
-#![warn(clippy::match_same_arms)]
-#![warn(clippy::mem_forget)]
-#![warn(clippy::mut_mut)]
-#![warn(clippy::mutex_integer)]
-#![warn(clippy::needless_continue)]
-#![warn(clippy::path_buf_push_overwrite)]
-#![warn(clippy::range_plus_one)]
-#![allow(clippy::cognitive_complexity)]
-#![allow(clippy::upper_case_acronyms)]
+#![deny(
+    rust_2018_idioms,
+    rust_2018_compatibility,
+    future_incompatible,
+    unused,
+    nonstandard_style
+)]
+#![deny(clippy::all, clippy::pedantic)]
```

Comment:
I'm personally not a big fan of globally enabling `clippy::pedantic`, given that there are several lints in that group that either have a lot of false positives or are generally just not very useful. I see that you've set some of them to `allow`, but I'd prefer to instead pick out the lints from that group that we like and specifically enable those.

Reply:
Re: `deny(clippy::all)`: I'm not against setting specific lints to `deny` that we really, never ever want to be allowed, but even though CI will fail if there are clippy warnings (IIRC), the reason I wouldn't want to `deny(clippy::all)` is because this will likely break compilation when new versions of Clippy are released with new lints in the `all` group, including for downstream packages which use oxipng as a library. I want old versions to still be compilable, even if they may emit warnings.